The following patches are taken directly from:

https://github.com/esnet/iperf/pull/664

as an upstream submission.

commit 9a66b3b0349e0a158bb4940b668a5cbc7c245762
Author: Philip Prindeville <philipp@redfish-solutions.com>
Date:   Tue Oct 31 13:44:34 2017 -0600

    Simplify endianness checks
    
    Linux can be built with too many types of C run-time library and it's
    not reasonable to have to enumerate all of them, especially since at
    least one of them (MUSL) goes out of its way to not be easily
    detectable.
    
    Instead, leverage autoconf better for Linux/BSD to detect either
    <endian.h> or <sys/endian.h> directly.
    
    Signed-off-by: Philip Prindeville <philipp@redfish-solutions.com>

diff --git a/configure.ac b/configure.ac
index f57e83f..cfb42ac 100644
--- a/configure.ac
+++ b/configure.ac
@@ -103,6 +103,14 @@ AC_CHECK_HEADERS([netinet/sctp.h],
 #endif
 ])
 
+AC_CHECK_HEADER([endian.h],
+		AC_DEFINE([HAVE_ENDIAN_H], [1], [Define to 1 if you have the <endian.h> header file.]),
+		AC_CHECK_HEADER([sys/endian.h],
+				AC_DEFINE([HAVE_SYS_ENDIAN_H], [1], [Define to 1 if you have the <sys/endian.h> header file.]),
+				AC_MSG_WARN([Couldn't find endian.h or sys/endian.h files: doing compile-time tests.])
+				)
+		)
+
 if test "x$with_openssl" = "xno"; then
     AC_MSG_WARN( [Building without OpenSSL; disabling iperf_auth functionality.] )
 else
diff --git a/src/iperf_config.h.in b/src/iperf_config.h.in
index bd03935..a9e51ec 100644
--- a/src/iperf_config.h.in
+++ b/src/iperf_config.h.in
@@ -15,6 +15,9 @@
 /* Define to 1 if you have the <dlfcn.h> header file. */
 #undef HAVE_DLFCN_H
 
+/* Define to 1 if you have the <endian.h> header file. */
+#undef HAVE_ENDIAN_H
+
 /* Have IPv6 flowlabel support. */
 #undef HAVE_FLOWLABEL
 
@@ -69,6 +69,9 @@
 /* Define to 1 if the system has the type `struct sctp_assoc_value'. */
 #undef HAVE_STRUCT_SCTP_ASSOC_VALUE
 
+/* Define to 1 if you have the <sys/endian.h> header file. */
+#undef HAVE_SYS_ENDIAN_H
+
 /* Define to 1 if you have the <sys/socket.h> header file. */
 #undef HAVE_SYS_SOCKET_H
 
